File: eecs268/2020fall/1pm/2020.09.04.inheritance-part-2/Dog.cpp
//Dog.cpp
#include "Dog.h"

int Dog::getAge() const
{
	return(age);
}

void Dog::doTrick()
{
	std::cout << "It's doing an amazing trick!\n";
}

void Dog::eat()
{
	std::cout << "Dog eating.\n";
}

void Dog::sleep()
{
	std::cout << "Dog sleeping.\n";
}

File: eecs268/2025spring/TR/2025.05.06.interfaces/animal-hierarchy/dog.py
#dog.py

from animal import Animal

class Dog(Animal):
    def __init__(self, age, breed):
        super().__init__(age)
        print('Dog created')
        self.breed = breed
        
    def do_trick(self):
        print('Dog doing amazing trick!')

    def sleep(self):
        print('Dog sleeping...')
        print('Hrp. Hrp HRP. Hrp hrrp.')
